server/app.js

```diff
@@ -31,8 +34,57 @@ const io = socketio(server, {
   },
 });
 
-io.on("connection", (socket) => {
-  console.log("connected");
+const loggedInUsers = new Map();
+
+io.use((socket, next) => {
+  if (socket.handshake.headers.cookie) {
+    const token = socket.handshake.headers.cookie.replace("token=", "");
+    jwt.verify(token, process.env.JWT_SECRET, (err, decoded) => {
+      if (err) {
+        socket.disconnect();
+        next();
+      }
+
+      const userExists = User.exists({ _id: decoded.id });
+      if (userExists) {
+        loggedInUsers.set(socket, decoded.id);
+        next();
+      } else {
+        socket.disconnect();
+        next();
+      }
+    });
+  } else {
+    socket.disconnect();
+    next();
+  }
+}).on("connection", (socket) => {
+  console.log(Array.from(loggedInUsers.values()));
+  socket.on("get unread notifications", (id) => {
+    Notification.find({ recipient: id, read: false }).then((notifications) => {
+      io.emit("new unread notifications", notifications);
```

Comment:
You only need to emit the notifications to the socket of the logged in user (currently we are emitting one user's notifications to all online users)